src/ex1/script.js

```diff
@@ -0,0 +1,129 @@
+const addTaskButton = document.querySelector(".add-task");
+const taskInput = document.querySelector("#task-text");
+const allTasksContainer = document.querySelector(".all-tasks-container");
+const pendingTasksText = document.querySelector(".pending-tasks");
+let pendingTasks = 0;
+const clearButton = document.querySelector(".clear-all");
+const sortButton = document.querySelector(".name-sort");
+
+// Event trigger when user clicks the add task button
+addTaskButton.addEventListener("click", () => {
+   addTask();
+});
+
+// Same as the button add task but only when user press "Enter"
+taskInput.addEventListener("keypress", (e) => {
+   if (e.key === "Enter") {
+      addTask();
+   }
+});
+
+// Clears all tasks when clear all button is clicked
+clearButton.onclick = () => {
+   document
+      .querySelectorAll(".task-container")
```

Comment:
consider: when using class-names / ids, it's better to have them in a separated constants javascript file and use them everywhere rather than having them hard coded multiple times in the code
```suggestion
      .querySelectorAll(TASKS_CONTAINER_SELECTOR)
```

Reply:
Makes a lot of sense. Added the file.